src/lashup_gm_mc.erl

```diff
@@ -194,10 +209,46 @@ forward_packet(MulticastPacket, Tree) ->
 
 bsend(MulticastPacket, Children) ->
   lists:foreach(fun (Child) ->
-    case erlang:send({?MODULE, Child}, MulticastPacket, [noconnect]) of
+    Begin = erlang:monotonic_time(),
+    try erlang:send({?MODULE, Child}, MulticastPacket, [noconnect]) of
       noconnect ->
         lager:warning("Dropping packet due to stale tree");
       _Result ->
-        ok
+        prometheus_counter:inc(
+          lashup, mc_send_bytes, [],
+          erlang:external_size(MulticastPacket))
+    after
+      prometheus_summary:observe(
+        lashup, mc_send_packets_seconds, [],
+        erlang:monotonic_time() - Begin)
     end
   end, Children).
+
+%%%===================================================================
+%%% Metrics functions
+%%%===================================================================
+
+-spec(init_metrics() -> ok).
+init_metrics() ->
+  prometheus_summary:new([
+    {registry, lashup},
+    {name, mc_do_multicast_seconds},
+    {duration_unit, seconds},
+    {help, "The time spent processing multicast packets."}
+  ]),
+  prometheus_summary:new([
+    {registry, lashup},
+    {name, mc_send_packets_seconds},
+    {duration_unit, seconds},
+    {help, "The time spent sending multicast packets."}
+  ]),
+  prometheus_counter:new([
+    {registry, lashup},
+    {name, mc_send_bytes},
+    {help, "Total number of multicast packets sent in bytes."}
+  ]),
+  prometheus_summary:new([
+    {registry, lashup},
+    {name, mc_receive_bytes},
```

Comment:
Similarly, `mc_receive_packets`

Reply:
Please see https://github.com/dcos/lashup/pull/49#discussion_r264088340

Reply:
Done, here are new metrics: `mc_receive_bytes_total` and `mc_receive_messages_total`.